text/0000-disconnect-flow.md

```diff
@@ -0,0 +1,134 @@
+- Feature Name: `disconnect_flow`
+- Start Date: 2021/06/02
+- RFC PR: 
+- Issue: [[Unity-Technologies/com.unity.multiplayer.mlapi#772](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/issues/772)
+
+# Summary
+
+This feature standardizes the flow for communicating disconnects, both between server and client, and between SDK and user.
+
+# Motivation
+
+The motivation is to better enable users to react to network events, including servers reacting to disconnecting clients, clients reacting to lost connections to the server, and clients reacting to other clients joining and leaving the game.
+
+# Guide-level explanation
+
+When network topology changes (via new or lost connections, whether intentionally or unintentionally), it's often desirable for the processes that are (or were) involved in the network to be informed of it so that they can react. In order to reactto these events, MLAPI provides the following callbacks that can be registered to NetworkManager:
+
+- `OnClientConnectedCallback` - Called on the **server** when a new client connection is **formed**. Expects a delegate with the following signature:
+  
+  ```C#
+  void OnClientConnected(int clientId){}
+  ```
+
+- `OnClientDisconnectCallback` - Called on the **server** when a client connection is **lost**. Expects a delegate with the following signature:
+  
+  ```C#
+  void OnClientDisconnected(int clientId, NetworkManager.DisconnectReason reason)
+  ```
+
+- `OnPeerConnectedCallback` - Called on the **client** when another client has **connected** to the same server. Expects a delegate with the following signature:
+  
+  ```C#
+  void OnPeerConnected(int clientId){}
```

Comment:
I love we're making the difference between client and peer. That's something that was pretty confusing the first time I looked into this